src/main/java/services/impl/QueryRequesterImpl.java

```diff
@@ -20,10 +23,19 @@
     public String makeQueryRequest(String uri, HttpHeaders headers) {
         RestTemplate restTemplate = new RestTemplate();
         ResponseEntity<String> response =
-                restTemplate.exchange(uri,
-                        HttpMethod.GET,
-                        new HttpEntity<String>(headers),
-                        String.class);
+                null;
+        try {
+            response = restTemplate.exchange(
+                    // Spring framework automatically encodes urls. This ensures we don't pass in an already
+                    // encoded url.
+                    URLDecoder.decode(uri, "UTF-8"),
```

Comment:
I'm not too familiar with this, but do you know where this is getting encoded in the first place? Is the `restTemplate.exchange` the thing that is re-encoding it? 
Is it possible to stop the original encoding from taking place?